modules/core/src/test/java/org/apache/ignite/logger/java/JavaLoggerTest.java

```diff
@@ -17,31 +17,154 @@
+
+    /**
+     * Check JavaLogger constructor from java.util.logging.config.file property
+     */
+    @Test
+    public void testDefaultConstructorWithProperty() throws Exception {
+        setJavaLoggerConfig();
+        IgniteLogger log1 = new JavaLogger();
+        IgniteLogger log2 = log1.getLogger(getClass());
+
+        assertTrue(log1.toString().contains("JavaLogger"));
+        assertTrue(log1.toString().contains(LOG_CONFIG_DEBUG));
+        assertTrue(log1.isDebugEnabled());
+
+        assertTrue(log2.toString().contains("JavaLogger"));
+        assertTrue(log2.toString().contains(LOG_CONFIG_DEBUG));
+        assertTrue(log1.isDebugEnabled());
+        System.clearProperty("java.util.logging.config.file");
+    }
+
+    /**
+     * Check JavaLogger constructor with custom logger
+     */
+    @Test
+    public void testDefaultConstructorWithCustomLogger() {
+        IgniteLogger log = new JavaLogger(Logger.getLogger(getClass().getSimpleName()));
+        assertTrue(log.toString().contains("JavaLogger"));
+        assertTrue(log.toString().contains("null"));
+    }
+
+    /**
+     * Check Grid logging
+     */
+    @Test
+    public void testGridLoggingWithDefaultLogger() throws Exception {
+        JavaLogger log = new JavaLogger();
+        assertTrue(log.isQuiet());
+        AtomicBoolean hasLog = new AtomicBoolean(false);
+        Filter filter = record -> {
+            String msg = record.getMessage();
+            if (msg.contains("JavaLogger") && msg.contains("config=") && msg.contains(JavaLogger.DFLT_CONFIG_PATH))
+                hasLog.set(true);
+            return true;
+        };
+
+        Logger rootLog = LogManager.getLogManager().getLogger("");
+        for (Handler hnd : rootLog.getHandlers()) {
+            if (hnd instanceof JavaLoggerFileHandler) {
+                hnd.setFilter(filter);
+                break;
+            }
+        }
+
+        IgniteConfiguration cfg = getConfiguration(getTestIgniteInstanceName());
+        cfg.setGridLogger(log);
+        startGrid(cfg);
+
+        doSleep(2_000);
+        assertTrue(hasLog.get());
+        stopAllGrids();
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testGridLoggingWithCustomLogger() throws Exception {
+        AtomicBoolean hasLog = new AtomicBoolean(false);
+        Filter filter = record -> {
+            if (record.getMessage().contains("JavaLogger") && record.getMessage().contains("config=null"))
+                hasLog.set(true);
+            return true;
+        };
+        ConsoleHandler hnd = new ConsoleHandler();
+        hnd.setFilter(filter);
+        hnd.setLevel(Level.INFO);
+
+        Logger rootLog = LogManager.getLogManager().getLogger("");
+        rootLog.addHandler(hnd);
+        JavaLogger log = new JavaLogger(rootLog);
+
+        IgniteConfiguration cfg = getConfiguration(getTestIgniteInstanceName());
+        cfg.setGridLogger(log);
+        startGrid(cfg);
+
+        doSleep(2_000);
```

Comment:
Please don't use sleep for waiting any events. Use instead GridTestUtils#waitForCondition.